app/routes/goal.py

```diff
@@ -0,0 +1,106 @@
+from flask import Blueprint, jsonify, request
+from app import db
+from app.models.goal import Goal
+from .routes_helper import validate_model
+from app.models.task import Task
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
```

Comment:
It's good that you're doing data validation here, but you should probably let the user know what is invalid.

Reply:
The reason it says "invalid data" is because the prewritten tests required it to be "Invalid data". However, I do agree that it should be more specific!